src/Underworld.ts

```diff
@@ -3776,7 +3776,7 @@ ${CardUI.cardListToImages(player.stats.longestSpell)}
           test_endCheckPromises();
         }
 
-        if (!effectState.shouldRefundLastSpell) {
+        if (!args.castForFree && !effectState.shouldRefundLastSpell) {
```

Comment:
Why `castForFree` instead of what we used to do with just charging the manacost when the spell was cast?

Reply:
This looked cleaner/simpler to me, since we can subtract a custom mana cost when we call castCards in the unit's file. Currently we have to store the mana before and then overwrite the mana after, and that seems prone to weird edge cases. I.E. if castCards triggers a series of events that changes the caster unit's mana, those changes would be overwritten after castCards resolves and we set `newMana = oldMana - customCost;`.

It also makes it easier to add other custom casting methods in the future. Maybe we want a unit to use its health or some other resource to cast a spell that would usually cost a player mana or something like that. Maybe a unit that doesn't have mana has some saved card action that it should be able to cast for free, or casts multiple cards together or something like that. It's easier to have a variable that says "ignore health, mana, cooldown, and other costs for this spell" and then define the special case in the unit file.